### PR TITLE
docs: Add info about auto-signoff

### LINKED ISSUE #8914: Add info about auto-signoff in the README
## Summary
We have a DCO check which runs on every PR to check if the commit has been signed off. 

Doing
```
git commit --amend --signoff
```
or something like 
```
git rebase HEAD~2 --signoff
```
is inconvenient. Signing off can be automated by adding the following hook in the local `.git` folder:
```bash
#!/bin/sh
#
# An example hook script to prepare the commit log message.
# Called by "git commit" with the name of the file that has the
# commit message, followed by the description of the commit
# message's source.  The hook's purpose is to edit the commit
# message file.  If the hook fails with a non-zero status,
# the commit is aborted.
#
# To enable this hook, rename this file to "prepare-commit-msg".

# This hook includes three examples.  The first comments out the
# "Conflicts:" part of a merge commit.
#
# The second includes the output of "git diff --name-status -r"
# into the message, just before the "git status" output.  It is
# commented because it doesn't cope with --amend or with squashed
# commits.
#
# The third example adds a Signed-off-by line to the message, that can
# still be edited.  This is rarely a good idea.

SOB=$(git var GIT_AUTHOR_IDENT | sed -n 's/^\(.*>\).*$/Signed-off-by: \1/p')
grep -qs "^$SOB" "$1" || echo "$SOB" >> "$1"
```
This needs be put in the `.git/hooks/prepare-commit-msg` file (you need to create a new file called `prepare-commit-msg`) and give it permission to execute using:
```
chmod +x .git/hooks/prepare-commit-msg
```

## TODO
- [ ] Add info about this in the README after https://github.com/keptn/keptn/blob/master/CONTRIBUTING.md#commit-types-and-scopes (create a heading called `Auto signoff commit messages` and add a line in https://github.com/keptn/keptn/blob/master/CONTRIBUTING.md#make-a-pull-request before `Finally, go to GitHub and create a Pull Request...` like this:
```
Make sure you sign-off your commits. To do this automatically check [this](https://github.com/keptn/keptn/blob/master/CONTRIBUTING.md#auto-signoff-commit-messages). Finally, go to GitHub and create a Pull Request.
```
- [ ] Raise a PR and get it merged

## Notes
- This issue was originally filed in https://github.com/keptn-sandbox/datadog-service/issues/82 but we are moving it here because we think having this in the keptn CONTRIBUTING would be better. 
- PR already raised in datadog-service by @nitinmewar: https://github.com/keptn-sandbox/datadog-service/pull/83/files

### PR DESCRIPTION
**Pull request by nitinmewar**
Signed-off-by: nitin mewar <nitinmewar28@gmail.com>

### Related Issues

Fixes #8914

### CODE REVIEW

**Comment by nitinmewar** [comment]
@vadasambar please review this.

**Review by vadasambar — APPROVED** [review]

**Comment by vadasambar** [comment]
LGTM. We need an approving review from someone who has write access to the repo. @keptn/maintainers

**Review by mowies — DISMISSED** [review]
I adjusted the wording a bit, but LGTM in general!
Thanks for you contribution!

**Comment by mowies** [comment]
@nitinmewar you could add yourself to the `CONTRIBUTORS.md` file :)

File: CONTRIBUTING.md
Comment (StackScribe) [inline CONTRIBUTING.md:273]:
```suggestion
This process is sometimes inconvenient but you can automate it
by creating a pre-commit git hook as follows:
```

File: CONTRIBUTING.md
Comment (StackScribe) [inline CONTRIBUTING.md:272]:
```suggestion
1. Create the hook:
```

File: CONTRIBUTING.md
Comment (StackScribe) [inline CONTRIBUTING.md:277]:
```suggestion
1. Add the following to the `prepare-commit-msg` file:
```

File: CONTRIBUTING.md
Comment (StackScribe) [inline CONTRIBUTING.md:283]:
```suggestion
1. Give it execution permissions by calling:
```

File: CONTRIBUTING.md
Comment (StackScribe) [inline CONTRIBUTING.md:258]:
```suggestion
We have a DCO check which runs on every PR to verify that the commit has been signed off.
```

**Review by StackScribe — COMMENTED** [review]
lgtm.  I made a couple editorial suggestions for you to consider

**Comment by mowies** [comment]
@nitinmewar don't worry, I will finish up with this :)

**Comment by mowies** [comment]
@nitinmewar should be all done now.
Can you double check that I added your name correctly to the contributors file?

**Comment by nitinmewar** [comment]
@mowies yes it's fine. thank you.

**Comment by mowies** [comment]
Perfect! Then i'll approve and merge this now :)
Thanks again for the contribution 🚀

**Review by mowies — APPROVED** [review]
LGTM

**Pull request merged**
